### PR TITLE
support expand-like reshape for kLoop/kInput in CPU backend

### PR DESCRIPTION
**Pull request by wyzero**

### CODE REVIEW

File: tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/util/env_var.h"
+
+namespace mlir_test {
+
+const std::string c_ft_path =
+    "tensorflow/compiler/mlir/disc/tests/regression/data/";
+
+TEST(ReshapeFusionCPUTest, kLoopTest) {
+  setenv("DISC_ENABLE_STITCH", "true", 1);
```
Comment (Orion34-lanbo) [inline tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc:23]: Just curious, have we support `stitch` fusion on cpu devices?

File: tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  setenv("DISC_ENABLE_SHAPE_CONSTRAINT_IR", "true", 1);
+  setenv("DISC_EXPECTED_KERNELS_IN_UT", "1", 1);
+  EXPECT_TRUE(feature_test_main(
```
Comment (Orion34-lanbo) [inline tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc:26]: How do we know the `expand-like` reshape op is fused in this test?

File: tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc:23]: Stitch on cpu is implemented on CPU in the end of 2021.  Here is an introducing material (start from page 16).
https://bladedisc.oss-cn-hangzhou.aliyuncs.com/docs/performance-optimization-practice.pdf

reshape like op support in stitch will be implemented in next PR.

File: tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc (hunk 2 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/tests/regression/reshape_fusion_cpu.cc:26]: line 25

**Review by Orion34-lanbo — APPROVED** [review]
LGTM

**Pull request merged**